src/pybind/mgr/dashboard/__init__.py

```diff
@@ -5,10 +5,35 @@
 """
 
 import os
+import re
+from typing import NamedTuple
 
 import cherrypy
 
-DEFAULT_VERSION = '1.0'
+DEFAULT_API_VERSION = '1.0'
+
+
+class APIVersion(NamedTuple):
+    major: int = 1
+    minor: int = 0
+
+    @classmethod
+    def from_string(cls, version_string):
+        result = re.match(r'application/vnd\.ceph\.api\.v(\d+)\.(\d+)\+json', version_string)
+        if result:
+            return cls(*(int(s) for s in result.groups()))
+        return None
+
+    def __str__(self):
+        return f'{self.major}.{self.minor}'
+
+    def handled_by(self, server_version):
+        return self.major == server_version.major and server_version.minor <= self.minor
+
+    @staticmethod
+    def to_string(version: str):
```

Comment:
This name is rather misleading ("to string" but returns an object). In fact if you add the type hints there, you'll notice:
```suggestion
    def to_string(version: str) -> 'APIVersion':
```

This method wouldn't be necessary if you store the default API version as a tuple, or even better, as an APIVersion named tuple. And you can inject the Class Variable (DEFAULT_API_VERSION) after creating the tuple):
```py
class APIVersion(NamedTuple):
  ...

# This way you have it properly namespaced
APIVersion.DEFAULT = APIVersion(1,0)
```

Reply:
Well I reasoned [here](https://github.com/ceph/ceph/pull/43285#discussion_r714697752) why it's not doable

Reply:
> This name is rather misleading ("to string" but returns an object). In fact if you add the type hints there, you'll notice:
> 
> This method wouldn't be necessary if you store the default API version as a tuple, or even better, as an APIVersion named tuple. And you can inject the Class Variable (DEFAULT_API_VERSION) after creating the tuple):
> 
> ```python
> class APIVersion(NamedTuple):
>   ...
> 
> # This way you have it properly namespaced
> APIVersion.DEFAULT = APIVersion(1,0)
> ```

Yeah I was also not sure abt the name. Maybe `to_object` then?